sr_robot_launch/scripts/sr_ur_arm_unlock

```diff
@@ -15,240 +15,11 @@
-        sleep_time = False
-        for arm in self._arms:
-            try:
-                headless_mode = rospy.get_param("/" + arm + "_sr_ur_robot_hw/headless_mode")
-            except KeyError:
-                headless_mode = False
-            if not headless_mode:
-                play_msg = self.call_arm_service(arm, "program_state", GetProgramState)
-                if ProgramState.STOPPED == play_msg.state.state or ProgramState.PAUSED == play_msg.state.state:
-                    rospy.loginfo("Not in headless mode. Starting program: %s for arm: %s",
-                                  play_msg.program_name, arm)
-                    self.call_arm_service(arm, "play", Trigger)
-                    sleep_time = True
-            else:
-                rospy.loginfo("Headless mode active, resending robot program to arm: %s", arm)
-                self.call_arm_service(arm, "resend_robot_program", Trigger, dashboard=False)
-                self.call_arm_service(arm, "resend_robot_program", Trigger, dashboard=False)
-                sleep_time = True
-        return sleep_time
-
-    def release_arm(self):
-        try:
-            rospy.loginfo("Unlock arm(s) signal received. Checking arms.")
-            rospy.loginfo("Checking e-stops ...")
-            self.check_arms_e_stops()
-            rospy.loginfo("Checking protective stops ...")
-            self.unlock_arms_if_protective_stop()
-            rospy.loginfo("Checking for faults ...")
-            if self.unlock_arms_if_fault():
-                rospy.loginfo("Resetting robot safety, please wait approximately 15 seconds...")
-                self.wait_for_mode('safety', SafetyMode.NORMAL, timeout=15)
-            rospy.loginfo("Checking protective stops again ...")
-            self.unlock_arms_if_protective_stop()
-            rospy.loginfo("Closing popups ...")
-            self.clear_arms_popups()
-            rospy.loginfo("Checking robot mode ...")
-            if self.check_arms_needs_starting():
-                self.startup_arms()
-            rospy.loginfo("Checking if program is loaded ...")
-            self.load_arms_program_if_unloaded()
-            rospy.loginfo("Checking if program is running ...")
-            self.start_arms_program_if_stopped()
-        except rospy.ServiceException as e:
-            for arm in self._arms:
-                rospy.logerr("Arm checking/restarting failed for arm: %s. %s", arm, e)
-
-    def brake_arm(self):
-        rospy.loginfo("Brake arm signal received.")
-        for arm in self._arms:
-            try:
-                self.call_arm_service(arm, "power_off", Trigger)
-            except rospy.ServiceException as e:
-                rospy.logerr("Arm braking failed for arm: %s. %s", arm, e)
+from sr_robot_launch.sr_ur_arm_unlock import SrUrUnlock
 
 
 if __name__ == "__main__":
     rospy.init_node('sr_ur_arm_unlock')
     sr_ur_unlock = SrUrUnlock()
-    rospy.spin()
+    while not rospy.is_shutdown():
```

Comment:
isnt putting rospy.spin() in while not rospy.is_shutdown redundant?

Reply:
71d2bc6d3854b4816d8790a2a364fe157085ec51